test/extended/networking/tuning.go

```diff
@@ -0,0 +1,211 @@
+
+		podDefinition := frameworkpod.NewAgnhostPod(namespace, podName, nil, nil, nil)
+		podDefinition.ObjectMeta.Annotations = map[string]string{"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s/%s", namespace, tuningNadName)}
+		pod := f.PodClient().Create(podDefinition)
+		err = frameworkpod.WaitForPodCondition(f.ClientSet, namespace, pod.Name, "Failed", 30*time.Second, func(pod *kapiv1.Pod) (bool, error) {
+			if pod.Status.Phase == kapiv1.PodPending {
+				return true, nil
+			}
+			return false, nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred(), "incorrect pod status")
+		o.Consistently(func() kapiv1.PodPhase {
+			pod, err := f.ClientSet.CoreV1().Pods(namespace).Get(context.Background(), podName, metav1.GetOptions{})
+			o.Expect(err).ToNot(o.HaveOccurred())
+			return pod.Status.Phase
+		}, 15*time.Second, 3*time.Second).Should(o.Equal(kapiv1.PodPending))
+	},
+		t.Entry("net.ipv4.conf.all.send_redirects", "net.ipv4.conf.all.send_redirects", "1"),
+		t.Entry("net.ipv4.conf.IFNAME.arp_filter", "net.ipv4.conf.IFNAME.arp_filter", "1"),
+	)
+
+	t.DescribeTable("pod sysctls should not affect node", func(sysctl, value, alternativeValue, path string) {
+		namespace := f.Namespace.Name
+		nodePod := frameworkpod.CreateExecPodOrFail(f.ClientSet, f.Namespace.Name, "nodeaccess-pod-", func(pod *kapiv1.Pod) {
+			pod.Spec.Volumes = []kapiv1.Volume{
+				{Name: "sysvolume", VolumeSource: kapiv1.VolumeSource{HostPath: &kapiv1.HostPathVolumeSource{Path: "/sys/class/net"}}},
+			}
+			pod.Spec.Containers[0].VolumeMounts = []kapiv1.VolumeMount{{Name: "sysvolume", MountPath: "/host/net"}}
+			pod.Spec.HostNetwork = true
+		})
+
+		nodeInterfaceNames, err := oc.AsAdmin().Run("exec").Args(nodePod.Name, "--", "ls", "/host/net").Output()
+		o.Expect(err).NotTo(o.HaveOccurred(), "unable to create nad")
+		nodeInterfaceName := strings.Fields(nodeInterfaceNames)[0]
+
+		nodeSysctlValue, err := oc.AsAdmin().Run("exec").Args(nodePod.Name, "--", "cat", fmt.Sprintf(path, nodeInterfaceName)).Output()
+		o.Expect(err).NotTo(o.HaveOccurred(), "unable to check sysctl")
+		sysctlValue := value
+		if sysctlValue == nodeSysctlValue {
+			sysctlValue = alternativeValue
+		}
+		err = createTuningNad(oc.AdminConfig(), namespace, tuningNadName, map[string]string{sysctl: sysctlValue})
+		o.Expect(err).NotTo(o.HaveOccurred(), "unable to create network-attachment-definition")
+
+		exutil.CreateExecPodOrFail(f.ClientSet, namespace, podName, func(pod *kapiv1.Pod) {
+			pod.ObjectMeta.Annotations = map[string]string{"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s/%s@%s", namespace, tuningNadName, nodeInterfaceName)}
+			pod.Spec.NodeName = nodePod.Spec.NodeName
+		})
+
+		result, err := oc.AsAdmin().Run("exec").Args(podName, "-n", namespace, "--", "cat", fmt.Sprintf(path, nodeInterfaceName)).Output()
+
+		o.Expect(err).NotTo(o.HaveOccurred(), "error checking pod sysctls")
+		o.Expect(result).To(o.Equal(sysctlValue), "incorrect sysctl value")
```

Comment:
no, I mean, the test case wants to be checking the value of the sysctl on the Multus-created interface (right?), but instead it is checking the value of the sysctl on the node interface again. Except, that shouldn't even work, because the path it tries to read from shouldn't even exist in the pod network namespace.

Reply:
Improved the test. Also added some By() messages to make the test better to read. 
In the test we do the following:
- create a pod (nodepod) to check the node sysctls. The pod has the node /proc dir mounted unter /host/proc
- check the nodepod sysctl value: ` cat "/host" + Sprintf(sysctlPath, ..)`
- create sysctl pod with sysctl value != that of the nodepod. Note that the annoation is: `namespace/name@IFNAME` , where IFNAME is the name of the node ifc, so we check NOT for `net1`, but for the same name as on the node
- check sysctll value on pod
- check that the node sysctl is not changed: ` cat "/host" + Sprintf(sysctlPath, ..)`